### PR TITLE
VACMS-6880: implement nonclinical services form design

### LINKED ISSUE #6880: Implement nonclinical services form design from #6879

### PR DESCRIPTION
**Pull request by JayDarnell**
## Description

Closes #6880 

## Testing done

Visual testing

## Screenshots

Register for Care
Screen Shot 2022-01-12 at 5 39 38 PM

Billing and Insurance
Screen Shot 2022-01-12 at 5 37 23 PM

Medical Records Offices
Screen Shot 2022-01-12 at 5 37 49 PM

## QA steps

As an admin user
- [x] Visit node/42302/edit and verify the page layout and register for care locations section reflect the UX design
- [x] Tinker with the settings in the related non-clinical service node by clicking the Edit location link and be sure the card updates to reflect the changes 
- [x] Visit node/42275/edit and verify the page layout and billing and insurance locations section reflect the UX design
- [x] Tinker with the settings in the related non-clinical service node by clicking the Edit location link and be sure the card updates to reflect the changes 
- [ ] Visit node/42290/edit and verify the page layout and medical records office locations section reflect the UX design
- [ ] Tinker with the settings in the related non-clinical service node by clicking the Edit location link and be sure the card updates to reflect the changes 

### Select Team for PR review

- [ ] `Platform CMS team`
- [ ] `Sitewide CMS Team`
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product Support`

### CODE REVIEW

**Comment by suzanne-gray** [comment]
@JayDarnell when I change a facility's hours to <do not specify hours> it displays like this. Would it make sense to suppress the hours display in this case?
Screen Shot 2022-01-12 at 3 14 21 PM

**Comment by suzanne-gray** [comment]
@JayDarnell it seems like `use facility hours` isn't working. No hours display here, but (I believe) it should say 24/7
Screen Shot 2022-01-12 at 3 43 54 PM

**Comment by suzanne-gray** [comment]
From /node/42302/edit (the system register for care page), if I click `edit location` then `add another service location` I get a confusing view: 
Screen Shot 2022-01-12 at 4 02 33 PM
Possible to have the `address` accordion automatically unfurled here? 
And/or add a heading "Location 1", "Location 2" above each address field? 
(maybe this is an enhancement?)

**Comment by suzanne-gray** [comment]
on /node/42999/ (facility register for care page) - When I edit a  location that does NOT use the facility address, the location address doesn't appear. I have to check and uncheck `use facility street address` for it to show up. (Mentioning this here but it may well be a separate bug.)
Edit location address

**Comment by JayDarnell** [comment]
> From /node/42302/edit (the system register for care page), if I click `edit location` then `add another service location` I get a confusing view: Screen Shot 2022-01-12 at 4 02 33 PM Possible to have the `address` accordion automatically unfurled here? And/or add a heading "Location 1", "Location 2" above each address field? (maybe this is an enhancement?)

I don't recall seeing this as a requirement within the original issue. If we find additional nice to haves lets collect them and put them into a separate issue.

**Comment by JayDarnell** [comment]
> @JayDarnell it seems like `use facility hours` isn't working. No hours display here, but (I believe) it should say 24/7 Screen Shot 2022-01-12 at 3 43 54 PM

Did you check to be sure the referenced facility has hours supplied?

**Comment by JayDarnell** [comment]
> @JayDarnell when I change a facility's hours to `do not specify hours` it displays like this. Would it make sense to suppress the hours display in this case? Screen Shot 2022-01-12 at 3 14 21 PM

This may require a change to the conditional views field logic. But yes it should be suppressed in this instance.

**Comment by EWashb** [comment]
So far I'm not seeing any issues with the card design. I did find (on the Billing and Insurance) that if you accidentally add an extra space in the city field, it mimics that space on the card. Not sure if we can prevent that. It's not a dealbreaker. 
Screen Shot 2022-01-13 at 9 50 17 AM

**Comment by EWashb** [comment]
A question after I thought about it. This only shows because I'm an admin, correct?
Screen Shot 2022-01-13 at 12 10 51 PM

**Comment by EWashb** [comment]
When attempting to add a new location for medical records, I am receiving this error. I'm choosing the correct facility.
Screen Shot 2022-01-13 at 12 18 58 PM

**Comment by JayDarnell** [comment]
I'm unable to locate the va/tests/revision-log failure @swirtSJW

**Review by swirtSJW — COMMENTED** [review]
Code looks good except for all the AWOL revision_log s.   WHenever adding content type ymls, `git add -p <filepath and name>` is the way to go.  The -p flag makes it so you interactively add each chunk .

File: config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml [hunk 1]
```diff
@@ -29,7 +29,6 @@ third_party_settings:
     group_edit:
       children:
         - moderation_state
-        - revision_log
```
[the remark this answers is not in the document]
Comment (JayDarnell) [reply config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml:32]: Bah, this guy again :)

**Review by swirtSJW — APPROVED** [review]
Looks good to me.

**Comment by swirtSJW** [comment]
Waiting to merge until after demo.

**Pull request merged**